src/Microsoft.AspNet.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -27,7 +27,7 @@ protected virtual void HandleNonHttpsRequest(AuthorizationContext filterContext)
             // body correctly.
             if (!string.Equals(filterContext.HttpContext.Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
             {
-                filterContext.Result = new HttpStatusCodeResult(StatusCodes.Status403Forbidden);
+                filterContext.Result = new HttpStatusCodeResult(StatusCodes.Status405MethodNotAllowed);
```

Comment:
I see your point. The correct scheme is required before selecting the correct method.